CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_CIReinforcementsManager.uc

```diff
@@ -0,0 +1,131 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  statusNone
+//  PURPOSE: Manager class to control the delay of any reinforcements issued
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class XComGameState_CIReinforcementsManager extends XComGameState_BaseObject;
+
+var array<DelayedReinforcementOrder> DelayedReinforcementOrders;
+
+var protectedwrite bool bNeedsUpdate;
+var const int Threshold;
+
+static function CreateReinforcementsManager()
+{
+	local XComGameState NewGameState;
+	local XComGameState_Player PlayerState;
+	local Object ThisObj;
+
+	if (GetReinforcementsManager(true) == none)
+	{
+		NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("CI: Creating reinforcements manager singleton");
+
+		NewGameState.CreateNewStateObject(class'XComGameState_CIReinforcementsManager');
+
+		`TACTICALRULES.SubmitGameState(NewGameState);
+	}
+
+	PlayerState = class'XComGameState_Player'.static.GetPlayerState(eTeam_XCom);
+	ThisObj = GetReinforcementsManager();
+	`XEVENTMGR.RegisterForEvent(ThisObj, 'PlayerTurnBegun', OnPlayerTurnBegun, ELD_OnStateSubmitted, , PlayerState);
+}
+
+function EventListenerReturn OnPlayerTurnBegun(Object EventData, Object EventSource, XComGameState GameState, Name Event, Object CallbackData)
+{
+	local XComGameState NewGameState;
+	local XComGameState_CIReinforcementsManager ManagerState;
+
+	NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("CI: Updating Reinforcements Manager");
+	ManagerState = GetReinforcementsManager();
+	ManagerState = XComGameState_CIReinforcementsManager(NewGameState.ModifyStateObject(class'XComGameState_CIReinforcementsManager', ManagerState.ObjectID));
+
+	ManagerState.bNeedsUpdate = true;
+
+	`TACTICALRULES.SubmitGameState(NewGameState);
+
+	return ELR_NoInterrupt;
+}
+
+static function XComGameState_CIReinforcementsManager GetReinforcementsManager(optional bool AllowNull = false)
+{
+	return XComGameState_CIReinforcementsManager(`XCOMHISTORY.GetSingleGameStateObjectForClass(class'XComGameState_CIReinforcementsManager', AllowNull));
+}
+
+function Update()
+{
+	local int idx;
+
+	for (idx = 0; idx < DelayedReinforcementOrders.Length; idx++)
+	{
+		if (DelayedReinforcementOrders[idx].TurnsUntilSpawn > 1)
+		{
+			DelayedReinforcementOrders[idx].TurnsUntilSpawn--;
+		}
+	}
+}
+
+function DelayedReinforcementOrder GetNextOrder()
```

Comment:
The function should return a `bool` and the `DelayedReinforcementOrder` should be actually returned with an `out` argument. This way you can differentiate if there is an order or if there is none qeued. See this for example:

https://github.com/WOTCStrategyOverhaul/CovertInfiltration/blob/a8c77e2903fc717ef8c59b83318f565a6e60bb90/CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_CovertActionExpirationManager.uc#L94